regional_mom6/regional_mom6.py

```diff
@@ -886,39 +896,54 @@ def bathymetry(
                 make topography (if true), or read an existing file.
             positivedown (Optional[bool]): If true, assumes that
                 bathymetry vertical coordinate is positive down.
+            chunks (Optional Dict[str, str]): Chunking scheme for bathymetry, eg {"lon": 100, "lat": 100}. Use lat/lon rather than the coordinate names in the input file.
 
 
         """
 
         if maketopog == True:
-            bathy = xr.open_dataset(
-                bathy_path, chunks={varnames["xh"]: 500, varnames["yh"]: 500}
-            )[varnames["elevation"]]
+            if chunks != "auto":
+                chunks = {varnames["xh"]: chunks["lon"], varnames["yh"]: chunks["lat"]}
+
+            bathy = xr.open_dataset(bathy_path, chunks=chunks)[varnames["elevation"]]
 
             bathy = bathy.sel(
-                {varnames["yh"]: slice(self.yextent[0] - 0.1, self.yextent[1] + 0.1)}
+                {varnames["yh"]: slice(self.yextent[0] - 1, self.yextent[1] + 1)}
             ).astype("float")
 
-            bathy = nicer_slicer(
-                bathy, np.array(self.xextent) + np.array([-0.1, 0.1]), varnames["xh"]
-            )
+            ## Here need to make a decision as to whether to slice 'normally' or with nicer slicer for 360 degree domain.
+
+            if bathy[varnames["xh"]][-1] - bathy[varnames["xh"]][0] > 355:
```

Comment:
```suggestion
            # if the grid goes around the globe then we want to use `nicer_slicer`
            if np.isclose(bathy[varnames["xh"]][-1] - bathy[varnames["xh"]][0], 360):
```

Reply:
This will need testing on an example notebook. I'd rather save it for another small PR later if there's an easy way to flag that?

Reply:
I don't understand why this will need testing, not that it's of course always good to have tests.

Reply:
This passes for gebco, but this function is also used for slicing surface forcing and the initial condition. In this implementation we're asserting the resolution that this function will work with. For example it fails if you use it on access om2 01, breaking the `model forced` notebook

Reply:
You are saying that a function called `bathymetry` is used to process surface forcing?

Reply:
Ohh no you're right this is the code that's calling nicer slicer, my bad sorry. But the point still stands that we are making an assumption about the bathymetry file. If we're happy that no-one would use bathymetry that's coarser than 10th degree then this does pass for Gebco

Reply:
Ok. But still you are saying that access-on2 bathymetry does not satisfy

```python 
np.isclose(bathy[varnames["xh"]][-1] - bathy[varnames["xh"]][0], 360, atol=1e-2)
```

I find this puzzling. How much is

```python 
bathy[varnames["xh"]][-1] - bathy[varnames["xh"]][0]
```
?

Reply:
I’m actually in disagreement with you in the sense that the code you suggest is making an assumption while what I suggest does not. If what I suggest makes an assumption the can you please explain me what the assumption is? If what we want is whether the bathymetry goes round the globe in longitude then I feel the least assumptions come from ensuring the longitude extent is 360 degrees.

Reply:
However, even if I'm not convinced, I'm not gonna step my foot down. Happy to keep it as you had it and I'll raise an issue to remind us to fix it?

For this PR we should definitely fix the dependencies-environment related issues, e.g., https://github.com/COSIMA/regional-mom6/pull/73/files#r1325511524

Reply:
If we change the tolerance to `1e-1` then it's ok for access om2 01. This is the longitude extent for it:

![image](https://github.com/COSIMA/regional-mom6/assets/53282288/f1c77a1e-6d55-4534-9bdd-4e91d1235351)

How about we change the tolerance to 0.1 (any resolution less than this would be silly for a regional model anyway) whereas `1e-2` would fail for other datasets that were coarser for 100th of a degree. It works for Gebco because it wraps around with a common start and end point of 180 / -180

I agree though that we should focus on getting these bigger fixes right in this PR and should do some nice quality of life PRs later once package is functioning and executing the example notebooks

Reply:
How about:

```suggestion
            if np.isclose(bathy[varnames["xh"]][-1] - bathy[varnames["xh"]][0] - horizontal_resolution, 360)
```

or whatever variable name the horizontal resolution is?

What you posted above suggests that I was forgetting that the two ends of a periodic grid are not exactly the same point (since that would be redundant) but they differ by a `Δx`.

Reply:
Why is this bathymetry on `xu_ocean` tho? Bathymetry lives on cell centers. So even in access-om2 I'd expect

```python
bathymetry['xt_ocean'][0] = -279.95
bathymetry['xt_ocean'][-1] = 80.05
```

right?

Reply:
Yeah that would work. You'd need to calculate the horizontal resolution for the given input bathymetry first via `bathy[varnames["xh"]][1] - bathy[varnames["xh"]][0]` or something since that's not stored

Reply:
```python
if np.isclose(bathy[varnames["xh"]][-1] - bathy[varnames["xh"]][1], 360)
```
?

Reply:
Or even cleaner

```python

horizontal_resolution = bathy[varnames["xh"]][1] - bathy[varnames["xh"]][0]

if np.isclose(bathy[varnames["xh"]][-1] - bathy[varnames["xh"]][0] - horizontal_resolution, 360)
```